examples/tasks/advanced/example_task.py

```diff
@@ -185,12 +185,12 @@ def _prompt_for_save(self):
 
     # Trait change handlers ------------------------------------------------
 
-    @on_trait_change("window:closing")
+    @observe("window:closing")
     def _prompt_on_close(self, event):
         """ Prompt the user to save when exiting.
         """
         close = self._prompt_for_save()
-        event.veto = not close
+        event.new.veto = not close
```

Comment:
assuming `event.new` is `window`

```suggestion
        window = event.new
        window.veto = not close
```